gh-actions/common/run-autopkgtest/action.yml

```diff
@@ -0,0 +1,172 @@
+    required: false
+    default: ''
+
+runs:
+  using: "composite"
+  steps:
+    - name: Check prerequisites
+      shell: bash
+      run: |
+        echo "::group::Check prerequisites"
+        set -eux
+
+        sudo apt install dctrl-tools
+        pkg_version=$(grep-dctrl -s Version -n - "${{ inputs.source-changes }}")
+
+        if [ -z "${pkg_version}" ]; then
+          echo "${{ inputs.source-changes }} is not a valid source .changes file" | \
+            tee "${GITHUB_STEP_SUMMARY}"
+          exit 1
+        fi
+
+        echo "PKG_VERSION=${pkg_version}" >> "${GITHUB_ENV}"
+
+        debs_dir=${{ inputs.debs-dir }}
+        if [ -z "$debs_dir" ]; then
+          debs_dir=$(dirname "${{ inputs.source-changes }}")
+        fi
+
+        if ls "${debs_dir}"/*_"${pkg_version}"_*.deb &>/dev/null; then
+          echo "DEBS_DIR=$debs_dir" >> "${GITHUB_ENV}"
+          echo "Testing debian packages"
+        else
+          msg="${debs_dir} is not containing any debian package"
+          echo "${msg}"
+
+          if [[ "$debs_dir" == "${{ inputs.debs-dir }}" ]]; then
+            echo "${msg}" >> "${GITHUB_STEP_SUMMARY}"
+            exit 1
+          fi
+
+          echo "The package source ${{ inputs.source-changes }} will be compiled before being tested."
+        fi
+
+        echo "::endgroup::"
+
+    - name: Setup LXD
+      uses: canonical/setup-lxd@v0.1.2
+      with:
+        channel: latest/stable
+
+    - name: Map the docker image name to LXD
+      shell: bash
+      run: |
+        echo "::group::Set image for autopkgtest"
+        set -eu
+
+        case "${{ inputs.lxd-image }}" in
+          ubuntu:latest|"")
+            lts=$(wget -O- https://api.launchpad.net/devel/ubuntu/series | \
+              jq -r '[.entries | .[] | select(.supported) | select(.version | test("^[0-9]?[02468].04")) | .name][0]')
+            img=ubuntu-daily:${lts}
+            ;;
+          ubuntu:rolling|"")
+            series=$(wget -O- https://api.launchpad.net/devel/ubuntu/series)
+            stable=$(echo "$series" | \
+              jq -r '.entries | .[] | select(.status=="Current Stable Release").name')
+            if [ -z "$stable" ]; then
+              stable=$(echo "$series" | jq -r '[.entries | .[] | select(.supported) | .name][0]')
+            fi
+            img=ubuntu-daily:${stable}
+            ;;
+          ubuntu:devel)
+            img=ubuntu-daily:devel
+            ;;
+          *)
+            img=$(echo "${{ inputs.lxd-image }}" | sed "s/^ubuntu:/ubuntu-daily:/")
+            ;;
+        esac
+
+        echo "AUTOPKGTESTS_BASE_IMAGE=${img}" >> "${GITHUB_ENV}"
+
+        echo "::endgroup::"
+
+    - name: Install autopkgtest
+      shell: bash
+      env:
+        TERM: dumb
+        DEBIAN_FRONTEND: noninteractive
+        DEBCONF_NONINTERACTIVE_SEEN: true
+      run: |
+        echo "::group::Install autopkgtest"
+        set -eu
+
+        sudo apt update
+        sudo apt install autopkgtest
```

Comment:
missing `-y`, no?

Reply:
Apparently no as it works :-D (I guess that's because is non interactive as per

```
        TERM: dumb
        DEBIAN_FRONTEND: noninteractive
        DEBCONF_NONINTERACTIVE_SEEN: true
```

But I can add it for consitency...